src/github.com/getlantern/errors/hide.go

```diff
@@ -0,0 +1,47 @@
+package errors
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/getlantern/hidden"
+)
+
+var (
+	hiddenErrors = make([]*structured, 100)
+	nextID       = uint64(0)
+	hiddenMutex  sync.RWMutex
+)
+
+// This trick saves the error to a ring buffer and embeds a non-printing
+// hiddenID in the error's description, so that if the errors is later wrapped
+// by a standard error using something like
+// fmt.Errorf("An error occurred: %v", thisError), we can subsequently extract
+// the error simply using the hiddenID in the string.
+func (e *structured) save() {
+	hiddenMutex.Lock()
+	b := make([]byte, 8)
+	binary.BigEndian.PutUint64(b, nextID)
+	e.id = nextID
+	e.hiddenID = hidden.ToString(b)
+	hiddenErrors[idxForID(nextID)] = e
+	nextID++
+	hiddenMutex.Unlock()
+}
+
+func get(hiddenID []byte) *structured {
+	id := binary.BigEndian.Uint64(hiddenID)
```

Comment:
It will panic if `hiddenID` is less than 8 bytes. Just a hint. Should be safe to ignore the check in our case.